### PR TITLE
Add connect/disconnectFromAstarte functions to public API

### PR DESCRIPTION
**Pull request by harlem88**
New functions in the AstarteDeviceSDK class:
- connectToAstarte
- disconnectFromAstarte

Signed-off-by: Antonio Gisondi <antonio.gisondi@secomind.com>

### CODE REVIEW

File: astarte-device-sdk/AstarteDeviceSDK.cpp [hunk 1]
```diff
@@ -373,3 +373,13 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
 
     return static_cast<AstarteDeviceSDK::ConnectionStatus>(m_astarteTransport->connectionStatus());
 }
+
+bool AstarteDeviceSDK::connectToBroker()
+{
+  return m_astarteTransport->connectToBroker();
```
Comment (rbino) [inline astarte-device-sdk/AstarteDeviceSDK.cpp:379]: `m_astarteTransport` is initialized asynchronously, so you should always check if it's `null` before calling methods on it

File: astarte-device-sdk/AstarteDeviceSDK.cpp [hunk 2]
```diff
@@ -373,3 +373,13 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
...
+}
+
+bool AstarteDeviceSDK::disconnectFromBroker()
+{
+  return m_astarteTransport->disconnectFromBroker();
```
Comment (rbino) [inline astarte-device-sdk/AstarteDeviceSDK.cpp:384]: See above

File: astarte-transport/astartetransport.cpp [hunk 3]
```diff
@@ -654,4 +654,14 @@ void AstarteTransport::routeWave(const Hyperspace::Wave &wave, int fd)
     Q_EMIT waveReceived(interface, w);
 }
 
+bool AstarteTransport::connectToBroker()
+{
+  return m_mqttBroker->connectToBroker();
```
Comment (rbino) [inline astarte-transport/astartetransport.cpp:659]: `m_mqttBroker` could be `null`, check with `isNull()` before calling methods on it

File: astarte-transport/astartetransport.cpp [hunk 4]
```diff
@@ -654,4 +654,14 @@ void AstarteTransport::routeWave(const Hyperspace::Wave &wave, int fd)
...
+}
+
+bool AstarteTransport::disconnectFromBroker()
+{
+  return m_mqttBroker->disconnectFromBroker();
```
Comment (rbino) [inline astarte-transport/astartetransport.cpp:664]: See above

**Review by rbino — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [1.0.1] - Unreleased
+### Added
+- Expose connect/disconnect broker methods in the SDK.
```
Comment (bettio) [inline CHANGELOG.md:9]: Let's separate Added and Changed with an empty line

File: astarte-device-sdk/AstarteDeviceSDK.cpp [hunk 6]
```diff
@@ -373,3 +373,21 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
 
     return static_cast<AstarteDeviceSDK::ConnectionStatus>(m_astarteTransport->connectionStatus());
 }
+
+bool AstarteDeviceSDK::connectToBroker()
+{
+  if (!m_astarteTransport) {
```
Comment (bettio) [inline astarte-device-sdk/AstarteDeviceSDK.cpp:379]: maybe we should add a qDebug().
`qDebug() << "Not yet initialized, cannot connect to broker."`

File: astarte-device-sdk/AstarteDeviceSDK.cpp [hunk 7]
```diff
@@ -373,3 +373,21 @@ AstarteDeviceSDK::ConnectionStatus AstarteDeviceSDK::connectionStatus() const
...
+    return false;
+  }
+
+  return m_astarteTransport->connectToBroker();
+}
+
+bool AstarteDeviceSDK::disconnectFromBroker()
+{
+  if (!m_astarteTransport) {
+    return false;
```
Comment (bettio) [inline astarte-device-sdk/AstarteDeviceSDK.cpp:389]: qDebug() << "Not initialized, cannot disconnect from broker."

File: astarte-device-sdk/AstarteDeviceSDK.h [hunk 8]
```diff
@@ -66,6 +66,9 @@ class AstarteDeviceSDK : public Hemera::AsyncInitObject
 
     ConnectionStatus connectionStatus() const;
 
+    bool connectToBroker();
+    bool disconnectFromBroker();
```
Comment (bettio) [inline astarte-device-sdk/AstarteDeviceSDK.h:70]: I'm not 100% sure if we should mention here `Broker` or just find a more generic naming, such as `connectToAstarte()`.

**Review by bettio — CHANGES_REQUESTED** [review]

**Review by bettio — APPROVED** [review]

**Pull request merged**